src/main/java/seedu/dukeofbooks/controller/LoanController.java

```diff
@@ -39,39 +32,42 @@ public void borrowBook(Person borrower, Book toBorrow,
         toBorrow.borrowItem();
     }
 
-    public void returnBook(Person borrower, Book toReturn)
+    public static void returnItem(LoanRecords loanRecords, Person borrower,
+                                  BorrowableItem toReturn)
             throws LoanRecordNotFoundException {
         if (!toReturn.isBorrowed()) {
             throw new LoanRecordNotFoundException("Book is not borrowed!");
         }
 
-        List<Loan> records = loanRecords.findByPersonItem(borrower, toReturn);
-        // TODO optimize find?
-        for (Loan loan : records) {
-            if (!loan.isReturned()) {
-                loan.setReturned(true);
-                toReturn.returnItem();
-            }
+        Loan loan = loanRecords.getLastActiveLoan(borrower);
+        if (loan == null) {
+            throw new LoanRecordNotFoundException("Cannot find an active loan!");
         }
+        loan.setReturned(true);
+        toReturn.returnItem();
     }
```

Comment:
To check again if the return updates the loan records stored in the person object

Reply:
I don't think so. Last time I asked our TA and he said it's fine to store all objects in one `LoanRecords`. So I think updating `LoanRecords` is enough, and when `Person` needs to check history, it just makes the query `loanRecords.findByPerson()` and it will return a list of past loan history.